### PR TITLE
Support stations with multiple z-levels

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
refactor: Map initialization now supports stations with multiple z-levels.
fix: The map reader no longer sometimes expands the world size inappropriately.
tweak: Pride's Mirror's destination has become less predictable.
/:cl:

Continuation of #34633, pursuant to completing the port of #32037. Big and probably deserves some actual scrutiny but things seem to be working in my tests and all the debug output checks out.

Done:
* **Support multi-z station maps**, planetstation here we come
* Load Lavaland and Reebe dynamically
* Allow any mix of multiple z-levels in one `.dmm` and multiple `.dmm` files
* Remove the last of the `ZLEVEL_` defines
* Allow map json files to specify traits for their maps
* Use traits to configure which levels are up/down relative to each other (ladders, chasms)
* Use traits to configure space linkage (transitions) rather than a separate config

Future work:
* Final tweaks to support multi-z mining zones and away missions
* Coordinate with DB folk to decide if default ordering of z-levels should be changed
* Migrate z-level lists in `GLOB` and subsystems to vars on `datum/space_level` for ease of maintenance
* Heap-allocation of z-levels and subregions, related deallocation and clearing

Output of "Debug Z-Levels" on a station map with three of them:
image

### CODE REVIEW

**Comment by Kromgar** [comment]
if this works you are a god. Will it support atmos across z-levels too?

**Comment by SpaceManiac** [comment]
Not in this PR but it's possible and planned and this is important groundwork for that.

**Comment by Bawhoppen** [comment]
Hey, so this is probably mostly unrelated, but if it would make code better in any way, we could remove zlevel randomization (torus or whatever we have nowadays) for space, since all our ruins are completely randomized in location anyways. I guess that would change GPS functionality a little, but probably no concern.

**Comment by Bawhoppen** [comment]
Only reason I mentioned it was because you did some other zlevel work like removing the defines.

**Comment by PKPenguin321** [comment]
[heavy breathing]

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
 
 #include "map_files\generic\CentCom.dmm"
-#include "map_files\generic\SpaceStation.dmm"
 
 #ifndef LOWMEMORYMODE
-
-	#include "map_files\generic\Space.dmm"
-	#include "map_files\generic\Space2.dmm"
-	#include "map_files\Mining\Lavaland.dmm"
-	#include "map_files\generic\City_of_Cogs.dmm"
```
Comment (Cyberboss) [inline _maps/_basemap.dm]: Move these down to all maps

**Review by Cyberboss — DISMISSED** [review]

File: _maps/_basemap.dm (hunk 1 above)
Comment (SpaceManiac) [reply _maps/_basemap.dm]: Added Lavaland and City_of_Cogs to the ALL_MAPS section.

**Comment by SpaceManiac** [comment]
The conflict is minor (#35291 editing the Lavaland bombcap multiplier), I'll take care of it and any others after the material content of the PR is approved.

**Comment by SpaceManiac** [comment]
Rebased in case anyone gets in a testmergey mood.

**Comment by Cyberboss** [comment]
It's been on for a while now, I'll give this a review and merge it soon

**Comment by AnturK** [comment]
Time to break everything.

**Pull request merged**